pymongo/pool.py

```diff
@@ -536,7 +536,10 @@ def ismaster(self, all_credentials=None):
 
     def _ismaster(self, cluster_time, topology_version,
                   heartbeat_frequency, all_credentials):
-        cmd = SON([('ismaster', 1)])
+        if self.opts.server_api:
```

Comment:
Does server_api get auto populated with a value above some `max_wire_version`? I am a bit confused as to how/if this change actually causes us to the hit the `hello` failPoint set in the updated spec tests?

Reply:
No, server_api is only set when it's passed explicitly to MongoClient. We have a test suite job that runs the entire test suite with server_api enabled. That is when we hit the "hello" failpoint. Grep for MONGODB_API_VERSION if you're curious.

Reply:
Ah, forgot about that. Thanks for clarifying!